x/staking/types/authz.go

```diff
@@ -46,8 +47,15 @@ func (authorization StakeAuthorization) MethodName() string {
 	return authzType
 }
 
+func (authorization StakeAuthorization) ValidateBasic() error {
+	if authorization.MaxTokens != nil && authorization.MaxTokens.IsNegative() {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidCoins, "negative coin amount: %v", authorization.MaxTokens)
+	}
+	return nil
```

Comment:
should we also validate that `AuthorizationType` is not `AUTHORIZATION_TYPE_UNSPECIFIED`?